packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/data/versioning/services/DefaultProjectDataVersiongRestService.java

```diff
@@ -167,4 +252,142 @@ public RestBranch getBranchById(IEditingContext editingContext, UUID branchId) {
     public RestBranch deleteBranch(IEditingContext editingContext, UUID branchId) {
         return null;
     }
+
+    private Optional<ResourceSet> getResourceSet(IEditingContext editingContext) {
+        return Optional.of(editingContext)
+            .filter(IEMFEditingContext.class::isInstance)
+            .map(IEMFEditingContext.class::cast)
+            .map(IEMFEditingContext::getDomain)
+            .map(AdapterFactoryEditingDomain::getResourceSet);
+    }
+
+    private Optional<EObject> createEmptyEObject(IEditingContext editingContext, String type) {
+        Optional<EObject> newEObjectOpt = this.getResourceSet(editingContext)
+            .map(ResourceSet::getPackageRegistry)
+            .map(EPackage.Registry::values)
+            .stream()
+            .flatMap(Collection::stream)
+            .filter(EPackage.class::isInstance)
+            .map(EPackage.class::cast)
+            .map(ePkg -> Optional.ofNullable(ePkg.getEClassifier(type))
+                    .filter(EClass.class::isInstance)
+                    .map(EClass.class::cast)
+                    .filter(eClass -> !(eClass.isAbstract() || eClass.isInterface())))
+            .flatMap(Optional::stream)
+            .map(EcoreUtil::create)
+            .findFirst();
+        return newEObjectOpt;
+    }
+
+    private void deserializePayload(IEditingContext editingContext, EObject eObject, Entry<String, Object> feature, Map<EObject, Map<String, Object>> emptyNewObjects, Set<EObject> newObjectsAttached) {
```

Comment:
Please let me know if this implementation should be moved into a dedicated class or anything else that could improve the readability or if something can be done to reduce the complexity.